sonic-post-syseeprom/scripts/post-syseeprom

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env python2
+
+'''
+    post-syseeprom
+    Syseeprom information gathering tool for SONiC
+    This tool will be started during the start phase of pmon container, gathering syseeprom info and write to state DB.
+    It's an one-shot task since syseeprom info are static.
+    With this tool, show syseeprom CLI will be able to get data from state DB instead of access hw or cache.
+'''
+
+try:
+    from sonic_daemon_base.daemon_base import DaemonBase
+    import argparse
+    from swsscommon import swsscommon
+except ImportError, e:
+    raise ImportError (str(e) + " - required module not found")
+
+PLATFORM_SPECIFIC_MODULE_NAME = 'eeprom'
+PLATFORM_SPECIFIC_CLASS_NAME = 'board'
+
+ERR_ARISTA_PLATFORM = 1
+ERR_FAILED_EEPROM = 2
+ERR_FAILED_UPDATE_DB = 3
+
+EEPROM_TABLE_NAME = 'EEPROM_INFO'
+
+def main():
+    parser = argparse.ArgumentParser(description='post-syseeprom tool')
+    parser.add_argument("-w", help="write syseeprom info into state DB", action='store_true')
+    parser.add_argument("-c", help="remove syseeprom info from state DB", action='store_true')
+    args = parser.parse_args()
+
+    Daemon_Base = DaemonBase()
+    (platform_path, hwsku_path) = Daemon_Base.get_path_to_platform_and_hwsku()
+
+    # Arista platform have their own implementation for eeprom tlv parsing, it's not inherited from eeprom base class.
+    # So for now this tool can not support Arista platform, but this will not cause anny issue on them. 
+    # decode-syseeprom scripts will cover Arista case, it will still read from hw/cache instead from DB on Arista platform.  
+    if 'arista' in platform_path:
+        Daemon_Base.log_warning('Arista platform does not support this yet')
+        return ERR_ARISTA_PLATFORM
+
+    if args.w:
+        eeprom_util = Daemon_Base.load_platform_util(PLATFORM_SPECIFIC_MODULE_NAME, PLATFORM_SPECIFIC_CLASS_NAME)
+        eeprom = eeprom_util.read_eeprom()
+        if eeprom is None :
+            Daemon_Base.log_error('Failed to read eeprom')
```

Comment:
@keboliu This is invalid logging API. Please use Logger from the sonic_daemon_base.daemon_base.

Reply:
To add to this, the DaemonBase class is intended to be inherited by the main class of a daemon. I think it's odd to instantiate an object of type DaemonBase just to call the methods within it.

I understand this is a one-shot application, not a daemon, but it might be better to inherit the DaemonBase class, since it doesn't currently have any dameon-specific functionality (forking, etc.). If we ever consider making DaemonBase do things like forking, then we can consider breaking out another subclass which handles logging, etc.

Reply:
I refactored this scripts with adding a class which inherits DaemonBase, and rewrite the log code with latest Logger class.